packages/react-components/src/collection/src/cancellablePromise.ts

```diff
@@ -0,0 +1,27 @@
+export interface CancellablePromise<T> {
+    promise: Promise<T>;
+    cancel(): void;
+}
+
+export interface CancellablePromiseError {
+    isCancelled: boolean;
+}
```

Comment:
The static Promise.reject function returns a Promise that is rejected. For debugging purposes and selective error catching, it is useful to make reason an instanceof Error.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject

Maybe we should reject a new type of error that has an extra property isCancelled. The type name could still be CancellablePromiseError , but it would be a real Error this time

Reply:
done